src/process_utility.c

```diff
@@ -233,11 +234,64 @@ check_alter_table_allowed_on_ht_with_compression(Hypertable *ht, AlterTableStmt
 			case AT_AddColumn:	 /* this is passed down */
 			case AT_ColumnDefault: /* this is passed down */
 			case AT_DropColumn:	/* this is passed down */
+
+			case AT_DropConstraint:
+			case AT_AlterConstraint:
 #if PG14_GE
 			case AT_ReAddStatistics:
 			case AT_SetCompression:
 #endif
 				continue;
+
+			/*
+			 * Conditionally allowed command. Adding a unique constraint
+			 * including non-segmentby column is not supported.
+			 */
+			case AT_AddConstraint:
+			{
+				Constraint *constraint = (Constraint *) cmd->def;
+
+				if (constraint->contype == CONSTR_UNIQUE)
+				{
+					List *allowed = NIL, *info = NIL;
+					ListCell *lc;
+					int i;
+
+					info = ts_hypertable_compression_get(ht->fd.id);
+
+					/* Unique constraints on segmentby columns are supported */
+					foreach (lc, info)
+					{
+						FormData_hypertable_compression *fd = lfirst(lc);
+						if (fd->segmentby_column_index > 0)
+							allowed = lappend(allowed,
+											  makeString(NameStr(fd->attname)));
+					}
+
+					/*
+					 * A unique constraint has to include dimention columns,
+					 * otherwise index creation is not possible.
+					 */
+					for (i = 0; i < ht->space->num_dimensions; i++)
+					{
+						Dimension *dim = &ht->space->dimensions[i];
+						allowed = lappend(allowed,
+										  makeString(NameStr(dim->fd.column_name)));
+					}
+
+					if(list_difference(constraint->keys, allowed) != NIL)
+						ereport(ERROR,
+								(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+								 errmsg("operation not supported on hypertables "
+										"that have compression enabled")));
+
+				}
+				else
+					continue;
+
+				break;
```

Comment:
Again, a codecov warning about this line. This is not important but having less of these warnings makes it easier to notice the important ones. Looks like the 'else/continue' can be removed.